File: cache-server/src/main/java/io/pivotal/demo/domain/WordCountWindowedCompsiteKey.java
package io.pivotal.demo.domain;

import java.util.Date;
import java.util.Objects;

public class WordCountWindowedCompsiteKey {
	public WordCountWindowedCompsiteKey(String word, Date start) {
		this.word = word;
		this.start = start;
	}

	public WordCountWindowedCompsiteKey() {
	}

	private String word;
	private Date start;

	public String getWord() {
		return word;
	}

	public void setWord(String word) {
		this.word = word;
	}

	public Date getStart() {
		return start;
	}

	public void setStart(Date start) {
		this.start = start;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		WordCountWindowedCompsiteKey that = (WordCountWindowedCompsiteKey) o;
		return Objects.equals(word, that.word) &&
				Objects.equals(start, that.start);
	}

	@Override
	public int hashCode() {
		return Objects.hash(word, start);
	}
}

File: web-ui/src/main/kotlin/io/pivotal/demo/domain/WordCountWindowedCompsiteKey.kt
package io.pivotal.demo.domain

import java.util.Date
import java.util.Objects

class WordCountWindowedCompsiteKey {

    var word: String? = null
    var start: Date? = null

    constructor(word: String, start: Date) {
        this.word = word
        this.start = start
    }

    constructor() {}

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as WordCountWindowedCompsiteKey?
        return word == that!!.word && start == that.start
    }

    override fun toString(): String {
        return "WordCountWindowedCompsiteKey{" +
                "word='" + word + '\''.toString() +
                ", start=" + start +
                '}'.toString()
    }

    override fun hashCode(): Int {
        return Objects.hash(word, start)
    }
}
